### ISSUE TITLE
JuicyPixels-3.2.5.2 fails to build on Windows with GHC 7.10.1

### ISSUE THREAD

**Issue by TerrorJack** [Twinside/Juicy.Pixels#99]
The error messages are listed below.

> Building JuicyPixels-3.2.5.2...
Preprocessing library JuicyPixels-3.2.5.2...
[ 1 of 30] Compiling Codec.Picture.Metadata.Exif ( src\Codec\Picture\Metadata\Exif.hs, dist\dist-sandbox-d06306b1\build\Codec\Picture\Metadata\Exif.o )
[ 2 of 30] Compiling Codec.Picture.Tiff.Types ( src\Codec\Picture\Tiff\Types.hs, dist\dist-sandbox-d06306b1\build\Codec\Picture\Tiff\Types.o )
[ 3 of 30] Compiling Codec.Picture.InternalHelper ( src\Codec\Picture\InternalHelper.hs, dist\dist-sandbox-d06306b1\build\Codec\Picture\InternalHelper.o )
[ 4 of 30] Compiling Codec.Picture.Types ( src\Codec\Picture\Types.hs, dist\dist-sandbox-d06306b1\build\Codec\Picture\Types.o )
ghc.exe: internal error: checkProddableBlock: invalid fixup in runtime linker: 000000000ef87910
    (GHC version 7.10.1 for x86_64_unknown_mingw32)
    Please report this as a GHC bug:  http://www.haskell.org/ghc/reportabug
This application has requested the Runtime to terminate it in an unusual way.
Please contact the application's support team for more information.

Can someone else with ghc 7.10.1 on 64-bit Windows reproduce the same problem?

**Comment by Twinside** [comment]
Well this looks like a GHC bug, it's even told in the output...

**Comment by TerrorJack** [comment]
I set up a clean Windows x64 environment, installed minghc 7.10.1, and JuicyPixels installed smoothly. I made a list of possible causes, but none proved to be right:

* Inappropriate package version. I compared the build logs and found that I was actually using zlib-0.5.4.2 (from a Stackage nightly config). So I started a new cabal sandbox and just typed `cabal install JuicyPixels`. Now it uses latest zlib, but the bug remains.

* Inappropriate binaries. I am using ghc & cabal binaries from www.haskell.org, but the mingw binaries are from msys2 (I removed mingw from the ghc binary package and made a symbolic link to use msys2). So I removed the symbolic link and instead used the mingw binaries provided in the ghc package. The bug remains.

Now I am trying to build ghc 7.10.1 using msys2 binaries and use the output build to install JuicyPixels. Fighting with build problems now, will post follow-up in a few hours.
